### DISCUSSION TITLE
Problem with Next Link and browser back button

### DISCUSSION THREAD

**Discussion by jakubhome** [vercel/next.js#69593] [App Router]
I have a problem in my NextJS application.
I have a ProductCard component that has a picture and a product name and is in the product listing.

The photo and product name is wrapped in a <Link>.

Moving from the listing to the product card, when I click on the photo or name, it correctly redirects me to the product page.

However, when I want to go back with the native browser button, the URL changes, but the page doesn't change.

When using router.push is same issue. 
In comparison, in classic <a> it works correctly

Any idea how to fix it?

**Comment by suprunchuk** [comment]
Try using `router.push` with the `shallow `option set to `false`. This will force a full page reload:
```js
import { useRouter } from 'next/router';

const ProductPage = () => {
  const router = useRouter();

  const handleGoBack = () => {
    router.push('/products', undefined, { shallow: false });
  };

  return (
    // ...
  );
};
```
